crates/sui-core/src/state_accumulator.rs

```diff
@@ -367,12 +367,14 @@ impl StateAccumulator {
         store: Arc<dyn AccumulatorStore>,
         epoch_store: &Arc<AuthorityPerEpochStore>,
     ) -> Self {
-        let chain = epoch_store.get_chain_identifier().chain();
-        if epoch_store.state_accumulator_v2_enabled() && chain != Chain::Mainnet {
-            StateAccumulator::V2(StateAccumulatorV2::new(store))
+        #[cfg(msim)]
+        if epoch_store.state_accumulator_v2_enabled() {
+            return StateAccumulator::V2(StateAccumulatorV2::new(store));
         } else {
-            StateAccumulator::V1(StateAccumulatorV1::new(store))
+            return StateAccumulator::V1(StateAccumulatorV1::new(store));
         }
+
+        StateAccumulator::V1(StateAccumulatorV1::new(store))
```

Comment:
can you use `if cfg!(msim)` so that we don't have an unreachable code warning in simtest

Reply:
Done